crates/sui-graphql-rpc/src/server/builder.rs

```diff
@@ -110,9 +142,14 @@ impl Server {
             })
         };
 
-        // Wait for both tasks to complete. This ensures that the service doesn't fully shut down
-        // until both the background task and the server have completed their shutdown processes.
-        let _ = join!(watermark_task, server_task);
+        // Wait for all tasks to complete. This ensures that the service doesn't fully shut down
```

Comment:
But in this case, watermark task never completes as it's in a loop?

Reply:
each task waits on the `cancellation_token` as well